addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -700,16 +701,17 @@ def write(self, vals):
             vals['date_last_stage_update'] = fields.Datetime.now()
         res = super().write(vals)
 
+        pools_to_update = set()
+
         for applicant in self:
-            if applicant.pool_applicant_id and applicant != applicant.pool_applicant_id and (not applicant.is_pool_applicant):
-                if 'email_from' in vals:
-                    applicant.pool_applicant_id.email_from = vals['email_from']
-                if 'partner_phone' in vals:
-                    applicant.pool_applicant_id.partner_phone = vals['partner_phone']
-                if 'linkedin_profile' in vals:
-                    applicant.pool_applicant_id.linkedin_profile = vals['linkedin_profile']
-                if 'type_id' in vals:
-                    applicant.pool_applicant_id.type_id = vals['type_id']
+            if not self.env.context.get('fields_synced', False) and applicant.pool_applicant_id:
+                pools_to_update.add(applicant.pool_applicant_id)
+
+        TO_SYNC_FIELDS = ('partner_name', 'email_from', 'partner_phone', 'linkedin_profile', 'categ_ids', 'type_id', 'availability', 'applicant_skill_ids')
```

Comment:
applicant_skill_ids is defined in hr_recruitment_skills, so you can't reference it here.
maybe define a method that returns the set of fields to sync, and override this method in hr_recruitment_skills